src/psyclone/tests/domain/lfric/arg_index_to_metadata_index_test.py

```diff
@@ -0,0 +1,175 @@
+    assert isinstance(cls._info, dict)
+    assert not cls._info
+
+
+def test_scalar():
+    '''Test the _scalar method. Note, we need at least one field/operator
+    for the metadata to be valid.
+
+    '''
+    meta_arg = ScalarArgMetadata("GH_REAL", "GH_READ")
+    meta_args = [
+        FieldArgMetadata("GH_REAL", "GH_INC", "W0"),
+        meta_arg]
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=meta_args)
+    cls = call_method("_scalar", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 1
+
+
+def test_field():
+    '''Test the _field method.'''
+    meta_arg = FieldArgMetadata("GH_REAL", "GH_INC", "W0")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_field", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+
+
+def test_field_vector():
+    '''Test the _field_vector method.'''
+    meta_arg = FieldVectorArgMetadata("GH_REAL", "GH_INC", "W0", "3")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_field_vector", meta_arg, metadata)
+    assert len(cls._info) == 3
+    assert cls._info[0] == 0
+    assert cls._info[1] == 0
+    assert cls._info[2] == 0
+
+
+def test_operator():
+    '''Test the _operator method.'''
+    meta_arg = OperatorArgMetadata("GH_REAL", "GH_WRITE", "W0", "W1")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_operator", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+
+
+def test_cma_operator():
+    '''Test the _cma_operator method.'''
+    meta_arg = ColumnwiseOperatorArgMetadata("GH_REAL", "GH_WRITE", "W0", "W1")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_cma_operator", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+
+
+def test_add_arg():
+    '''Test the _add_arg utility method.'''
+    meta_arg = FieldArgMetadata("GH_REAL", "GH_WRITE", "W0")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_add_arg", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+    assert cls._index == 1
+
+
+def test_remaining_methods():
+    '''Test the remaining methods. Each of these increments the _index
+    variable by one, apart from _cell_map which increments it by four.
+
+    '''
+    cls = ArgIndexToMetadataIndex
+    cls._initialise(None)
+    count = 0
+    for method_name in (
+            ["_cell_position", "_mesh_height", "_mesh_ncell2d_no_halos",
+             "_mesh_ncell2d"]):
+        getattr(cls, method_name)()
+        count += 1
+        assert cls._index == count
+    cls._cell_map()
+    count += 4
+    assert cls._index == count
+    for method_name in (
```

Comment:
Is there some reason these have to be done *after* cell_map? If so, please add a comment, if not, please merge with the loop at L160.

Reply:
Yes, they require an argument. I've added comments.